hazelcast/src/main/java/com/hazelcast/concurrent/lock/LockEvictionProcessor.java

```diff
@@ -59,27 +62,28 @@ public void process(EntryTaskScheduler<Data, Integer> scheduler, Collection<Sche
     }
 
     private void sendUnlockOperation(Data key, int version) {
-        UnlockOperation operation = new UnlockIfLeaseExpiredOperation(namespace, key, version);
-        try {
-            submit(operation, key);
-        } catch (Throwable t) {
-            logger.warning(t);
-        }
-    }
+        InternalPartitionService partitionService = nodeEngine.getPartitionService();
+        int partitionId = partitionService.getPartitionId(key);
+        InternalPartition partition = partitionService.getPartition(partitionId);
+        Address owner = partition.getOwnerOrNull();
+        Address thisAddress = nodeEngine.getThisAddress();
+        boolean runningOnOwner = thisAddress.equals(owner);
```

Comment:
partition ownership should be checked in partition thread. you can move this logic inside the `UnlockIfLeaseExpiredOperation`.